lightning-invoice/src/de.rs

```diff
@@ -562,17 +570,22 @@ impl FromBase32 for Fallback {
 				if bytes.len() != 20 {
 					return Err(ParseError::InvalidPubKeyHashLength);
 				}
-				//TODO: refactor once const generics are available
-				let mut pkh = [0u8; 20];
-				pkh.copy_from_slice(&bytes);
+				let pkh = match PubkeyHash::from_slice(&bytes) {
+					Ok(pkh) => pkh,
+					Err(_) => return Err(ParseError::InvalidPubKeyHash),
+				};
```

Comment:
`from_slice` will check the length, so we can remove our explicit check in favor of the new one. Probably could have the `match` pattern on the exact error variant. That way we won't need to add a new error variant and we'll fail to compile if new variants are added upstream.

Reply:
nice, thank you this worked out great